### PR TITLE
Update the no sites followed empty state

### LINKED ISSUE #14006: Update UI of the empty view on the following tab
Update the UI of the empty view on the following tab.

Consider if we need to make any changes to the design when
- the user is logged in with a self hosted site
- the user is logged in with .com account
- the user is logged in with both self-hosted site and .com account

Note:
- The following design might be out of date. Check out figma to see the most current design.
Screenshot 2020-04-27 at 12 12 33

### PR DESCRIPTION
**Pull request by emilylaguna**
Fixes #14006

Updates the following empty state

###  Light Mode
|Self Hosted|WP.com|
|---|---|
|Simulator Screen Shot - iPhone 11 Pro (13 3) - 2020-05-27 at 17 47 55|Simulator Screen Shot - iPhone 11 Pro (13 3) - 2020-05-27 at 17 46 40|

###  Dark Mode
|Self Hosted|WP.com|
|---|---|
|Simulator Screen Shot - iPhone 11 Pro (13 3) - 2020-05-27 at 17 47 56|Simulator Screen Shot - iPhone 11 Pro (13 3) - 2020-05-27 at 17 53 44|

### Add Sites (Logged in only)
|Light Mode|Dark Mode|
|---|---|
|Simulator Screen Shot - iPhone 11 Pro (13 3) - 2020-05-27 at 17 56 15|Simulator Screen Shot - iPhone 11 Pro (13 3) - 2020-05-27 at 17 56 13|

### Add Sites from cog (Logged in only)
|Dark Mode|Light Mode|
|---|---|
|Simulator Screen Shot - iPhone 11 Pro (13 3) - 2020-05-27 at 17 56 38|Simulator Screen Shot - iPhone 11 Pro (13 3) - 2020-05-27 at 17 56 39|

### WP.com with no followed sites
1. Login to a WP.com site that's following no sites
2. Open the Reader 
3. Tap the Following tab
4. Verify it looks good in both dark and light modes and you see the 'Discover Sites' button
5. Tapping on the button should bring you to the add sites view
6. Verify the empty message on the add sites view looks good and has no button (you're already on that view)
7. Back out and tap the Cog, then tap 'followed sites' tab
8. Verify the empty message on the add sites view looks good and has no button (you're already on that view)

### Self hosted
1. Logout
2. Login using a self hosted site
3. Open the Reader 
4. Tap the Following tab
5. Verify it looks good in both dark and light modes and you don't see the `Discover Sites` button
5. Tapping on the button should bring you to the add sites view

## PR submission checklist:

- [X] I have considered adding unit tests where possible.
- [X] I have considered adding accessibility improvements for my changes.
- [X] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by leandroalonso** [comment]
> Tapping on the button should bring you to the add sites view

Any specific reason why we're pushing that view instead of presenting it?

File: WordPress/Classes/ViewRelated/Reader/ReaderFollowedSitesViewController.swift [hunk 1]
```diff
@@ -166,12 +167,12 @@ class ReaderFollowedSitesViewController: UIViewController, UIViewControllerResto
         }
 
         currentKeyboardHeight = keyboardFrame.height
-        configureNoResultsView()
+//        configureNoResultsView()
```
Comment (leandroalonso) [inline WordPress/Classes/ViewRelated/Reader/ReaderFollowedSitesViewController.swift:170]: We may want to remove this?

File: WordPress/Classes/ViewRelated/Reader/ReaderFollowedSitesViewController.swift [hunk 2]
```diff
@@ -166,12 +167,12 @@ class ReaderFollowedSitesViewController: UIViewController, UIViewControllerResto
...
     }
 
     @objc func keyboardWillHide(_ notification: Foundation.Notification) {
         currentKeyboardHeight = 0
-        configureNoResultsView()
+//        configureNoResultsView()
```
Comment (leandroalonso) [inline WordPress/Classes/ViewRelated/Reader/ReaderFollowedSitesViewController.swift:175]: And this?

File: WordPress/Classes/ViewRelated/Reader/ReaderFollowedSitesViewController.swift (hunk 1 above)
Comment (emilylaguna) [reply WordPress/Classes/ViewRelated/Reader/ReaderFollowedSitesViewController.swift:170]: 🤦 oops.

**Comment by emilylaguna** [comment]
> Any specific reason why we're pushing that view instead of presenting it?

I didn't change how the button worked, but I looked into it a bit and it's pretty old code. I'm assuming the view relies on the viewWill/viewDid methods to refresh/resync etc.

**Pull request merged**